src/main/java/uk/gov/ons/census/casesvc/service/InvalidAddressService.java

```diff
@@ -42,6 +41,52 @@ public void processMessage(ResponseManagementEvent invalidAddressEvent) {
         "Invalid address",
         EventType.ADDRESS_NOT_VALID,
         invalidAddressEvent.getEvent(),
-        convertObjectToJson(invalidAddressEvent.getPayload().getInvalidAddress()));
+        convertObjectToJson(invalidAddress));
+  }
+
+  private boolean processEvent(ResponseManagementEvent addressEvent) {
+    String logEventDescription;
+    EventType logEventType;
+    String logEventPayload;
+    EventDTO event = addressEvent.getEvent();
+
+    switch (event.getType()) {
+      case ADDRESS_NOT_VALID:
+        return true;
+
+      case ADDRESS_MODIFIED:
+        logEventDescription = "Address modified";
+        logEventType = EventType.ADDRESS_MODIFIED;
+        logEventPayload = addressEvent.getPayload().getAddressModification();
+        break;
+
+      case ADDRESS_TYPE_CHANGED:
+        logEventDescription = "Address type changed";
+        logEventType = EventType.ADDRESS_TYPE_CHANGED;
+        logEventPayload = addressEvent.getPayload().getAddressTypeChange();
+        break;
+
+      case NEW_ADDRESS_REPORTED:
+        logEventDescription = "New Address reported";
+        logEventType = EventType.NEW_ADDRESS_REPORTED;
+        logEventPayload = addressEvent.getPayload().getNewAddressReported();
+        break;
+
+      default:
+        // Should never get here
+        throw new RuntimeException(String.format("Event type '%s' not found", event.getType()));
```

Comment:
Strictly speaking it's "Unexpected event type X on topic"

Reply:
changed